### ISSUE TITLE
1x3 furnitures placed wrongly

### ISSUE THREAD

**Issue by Ecilpse** [PluginBugs/Issues-ItemsAdder#3394]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I am sure this is a bug and it is not caused by a misconfiguration or by another plugin.
- [X] I already searched on this [Github page](https://github.com/PluginBugs/Issues-ItemsAdder/issues) to check if the same issue was already reported.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if a solution is already known.
- [X] I already searched on the [forums](https://forum.devs.beer/) to check if anyone already has a solution for this.

### Discord tag (optional)

EclipseMC#oing1215

### What happened?

![2024-02-03_12 44 37](https://github.com/PluginBugs/Issues-ItemsAdder/assets/25452835/cfde1e9c-5ca6-4819-a29e-a048b4271dd3)
![2024-02-03_12 44 39](https://github.com/PluginBugs/Issues-ItemsAdder/assets/25452835/96442360-2140-47e6-aa9a-f4716d6a8825)

Hitbox barrier are placed wrongly depending on the direction 1x3 furniture is placed,
and player cannot sit in one of the three hitboxes.

### Steps to reproduce the issue
```
  #bamboo
  furniture_bamboo_bench:
    display_name: 대나무 벤치 
    resource:
      material: LEATHER_HORSE_ARMOR
      generate: false
      model_path: funitures/bamboo_bench
    behaviours:
      furniture:
        fixed_rotation: true
        entity: item_frame
        gravity: false
        small_hitbox: false
        solid: true
        sit_all_solid_blocks: true
        placeable_on:
          floor: true
          ceiling: false
          walls: false
        hitbox:
          length: 1
          width: 3
          height: 1
          width_offset: 0.5
        sound:
          place:
            name: block.stone.place
          break:
            name: block.stone.break
      furniture_sit:
        sit_height: 1.1
```

### Server version

PURPUR-1.20.2-RO.1

### ItemsAdder Version

3.6.3-beta-9

### ProtocolLib Version

5.1.1-SNAPSHOT-673

### LoneLibs Version

1.0.45

### Screenshots/Videos (you can drag and drop files or paste links)

video: https://youtu.be/Mebg88FMCDo

**Comment by ShizuArt** [comment]
Same issue for me but only with the hitbox. The direction doesn't affect it (with `item_frame` entity there is no issue)

**Server version**
Paper version git-Paper-196 (MC: 1.20.1) (Implementing API version 1.20.1-R0.1-SNAPSHOT)

**ItemsAdder Version**
3.6.3-beta-12

**ProtocolLib Version**
5.2.0-SNAPSHOT-679

**LoneLibs Version**
1.0.45


```
  jardin_bain_soleil_ouvert:
    enabled : true
    permission: ""
    display_name: "<#5eb054>Bain de soleil"
    behaviours:
      furniture:
        entity: armor_stand
        small: true
        solid : true
        fixed_rotation: true
        placeable_on :
          floor: true
          ceiling: false
          walls: false
        opposite_direction: true
        hitbox:
          height: 1
          lenght: 1
          width: 2
          width_offset: -0.5
        sound:
          place:
            name: block.wood.place
          break:
            name: block.wood.break
      furniture_sit:
        sit_height: 0.5
    resource:
      material: PAPER
      generate: false
      model_path: jardin/jardin_bain_soleil_ouvert
      model_id: 1990065
```
With `opposite_direction: true` 
![image](https://github.com/PluginBugs/Issues-ItemsAdder/assets/98397139/3c4b8386-c687-4164-8af7-f90d53382b28)

With `opposite_direction: false`
![image](https://github.com/PluginBugs/Issues-ItemsAdder/assets/98397139/7764fdc2-3102-46f2-beee-51e39055016e)

With `entity: item_frame`
![image](https://github.com/PluginBugs/Issues-ItemsAdder/assets/98397139/997ceaaf-0c3b-4366-86f7-fb4f800bf636)

**Comment by LoneDev6** [comment]
Can you confirm that you read the tutorial?
https://itemsadder.devs.beer/plugin-usage/adding-content/furniture/furniture-collisions#wrong-hitbox-location

The hitboxes where heavily tested and lot of artists are creating packs, none of them have this issue so I think it's a misconfiguration.
Please let me know.

**Comment by ShizuArt** [comment]
You are right, my problem was that using "opposite_direction" necessarily requires modifying the offset since it inverts the position model :3 I'm a little stupid sometimes
Are there plans in the future to be able to customize the location (x;y;z) of barriers like Oraxen or Crucible does?
Because for example in my Easter pack, I made an arch
It's impossible with AI to make a hitbox that follows the arch and leaves a passage in the middle

**Comment by LoneDev6** [comment]
> I'm a little stupid sometimes
Nah you're not, my system is stupid. I coded it long time ago because it was the fastest way to easily add support to any model, but I understand it is quite confusing at times.

> Are there plans in the future to be able to customize the location (x;y;z) of barriers
I can't find any other suggestion on Github so I opened one, I will maybe implement that in the upcoming updates.
https://github.com/PluginBugs/Issues-ItemsAdder/issues/3473
Thanks for the suggestion.
